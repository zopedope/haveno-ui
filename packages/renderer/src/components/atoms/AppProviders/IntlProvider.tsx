// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
