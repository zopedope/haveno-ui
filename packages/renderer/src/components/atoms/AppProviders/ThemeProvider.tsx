// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
