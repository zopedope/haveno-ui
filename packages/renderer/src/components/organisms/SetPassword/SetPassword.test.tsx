// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SetPassword } from ".";
import { AppProviders } from "@atoms/AppProviders";

describe("organisms::SetPassword", () => {
  it("renders without exploding", () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();
    const { asFragment, unmount } = render(
      <AppProviders>
        <SetPassword onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("calls onGoBack", async () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();
    const { unmount } = render(
      <AppProviders>
        <SetPassword onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    expect(onBackSpy).to.not.toHaveBeenCalled();
    fireEvent.click(await screen.findByLabelText("Click to go back"));
    expect(onBackSpy).to.toHaveBeenCalledTimes(1);
    expect(onNextSpy).to.not.toHaveBeenCalled();
    unmount();
  });

  it("blocks submit if validation fails", async () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();

    const { unmount } = render(
      <AppProviders>
        <SetPassword onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    expect(onNextSpy).to.not.toHaveBeenCalled();
    fireEvent.click(await screen.findByLabelText("Click to submit"));
    expect(onNextSpy).to.not.toHaveBeenCalled();
    unmount();
  });

  it("calls onSubmit if validation succeeds", async () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();
    const user = userEvent.setup();
    const { unmount } = render(
      <AppProviders>
        <SetPassword onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    await user.type(screen.getByLabelText("Enter password"), "Qwe$9999", {
      skipAutoClose: true,
    });
    await user.type(screen.getByLabelText("Repeat password"), "Qwe$9999", {
      skipAutoClose: true,
    });
    expect(onNextSpy).to.not.toHaveBeenCalled();
    fireEvent.submit(screen.getByLabelText("Click to submit"));
    expect(onNextSpy).to.toHaveBeenCalledTimes(1);
    unmount();
  });
});
