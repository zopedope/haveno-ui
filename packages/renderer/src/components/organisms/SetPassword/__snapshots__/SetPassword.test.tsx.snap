// Vitest Snapshot v1

exports[`organisms::SetPassword > renders without exploding 1`] = `
<DocumentFragment>
  <form
    data-testid="change-password-form"
  >
    <div
      class="mantine-Stack-root mantine-lfk3cq"
    >
      <div
        class="mantine-Container-root mantine-15pjuqq"
      >
        <h1
          class="mantine-Title-root mantine-10yhpe6"
        >
          Create password
        </h1>
      </div>
      <div
        class="mantine-Text-root mantine-1xgwjc0"
      >
        All your data is stored locally on your machine. Haveno uses solely a password.
      </div>
      <div
        class="mantine-TextInput-root mantine-1u48g6v"
      >
        <label
          class="mantine-TextInput-label mantine-1bjo575"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            aria-label="Enter password"
            class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-170eizj"
            id="password"
            tabindex="1"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="mantine-TextInput-root mantine-1u48g6v"
      >
        <label
          class="mantine-TextInput-label mantine-1bjo575"
          for="repeatPassword"
          id="repeatPassword-label"
        >
          Repeat password
        </label>
        <div
          class="mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            aria-label="Repeat password"
            class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-170eizj"
            id="repeatPassword"
            tabindex="2"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="mantine-63n06h"
      />
      <div
        class="mantine-Group-root mantine-19jxmdp"
      >
        <button
          aria-label="Click to go back"
          class="mantine-UnstyledButton-root mantine-Group-child mantine-7r9su"
          type="button"
        >
          <span
            class="mantine-Text-root mantine-1g5fxnv"
          >
            Go Back
          </span>
        </button>
        <button
          aria-label="Click to submit"
          class="mantine-Button-filled mantine-Button-root mantine-Group-child mantine-18llzhv"
          type="submit"
        >
          <div
            class="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              class="mantine-qo1k2 mantine-Button-label"
            >
              Next
            </span>
          </div>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
